convert/convert.go

```diff
@@ -0,0 +1,289 @@
+package convert
+
+import (
+	"bytes"
+	"cmp"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"slices"
+
+	"github.com/mitchellh/mapstructure"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/jmorganca/ollama/convert/sentencepiece"
+	"github.com/jmorganca/ollama/llm"
+)
+
+type Params struct {
+	VocabSize        int     `json:"vocab_size"`
+	HiddenSize       int     `json:"hidden_size"`       // n_embd
+	HiddenLayers     int     `json:"num_hidden_layers"` // n_layer
+	ContextSize      int     `json:"max_position_embeddings"`
+	IntermediateSize int     `json:"intermediate_size"`
+	AttentionHeads   int     `json:"num_attention_heads"` // n_head
+	KeyValHeads      int     `json:"num_key_value_heads"`
+	NormEPS          float64 `json:"rms_norm_eps"`
+	RopeFreqBase     float64 `json:"rope_theta"`
+	BoSTokenID       int     `json:"bos_token_id"`
+	EoSTokenID       int     `json:"eos_token_id"`
+}
+
+type MetaData struct {
+	Type    string `mapstructure:"dtype"`
+	Shape   []int  `mapstructure:"shape"`
+	Offsets []int  `mapstructure:"data_offsets"`
+}
+
+func ReadSafeTensors(fn string, offset uint64) ([]llm.Tensor, uint64, error) {
+	f, err := os.Open(fn)
+	if err != nil {
+		return []llm.Tensor{}, 0, err
+	}
+	defer f.Close()
+
+	// read the first 8 bytes of the file
+	buf := make([]byte, 8)
+	_, err = io.ReadFull(f, buf)
+	if err != nil {
+		return []llm.Tensor{}, 0, err
+	}
+
+	// convert to little endian uint64
+	num := binary.LittleEndian.Uint64(buf)
+
+	buf = make([]byte, num)
+	_, err = io.ReadFull(f, buf)
+	if err != nil {
+		return []llm.Tensor{}, 0, err
+	}
+
+	d := json.NewDecoder(bytes.NewBuffer(buf))
+	d.UseNumber()
+	var parsed map[string]interface{}
+	if err = d.Decode(&parsed); err != nil {
+		return []llm.Tensor{}, 0, err
+	}
+
+	var keys []string
+	for k := range parsed {
+		keys = append(keys, k)
+	}
+
+	slices.SortFunc(keys, func(a, b string) int {
```

Comment:
This looks like a lexicographic sort? If that's the case we can just use the default slices sort:
```
slices.Sort(keys) // This sorts the slice lexicographically
```